OlinkAnalyze/R/Olink_one_non_parametric.R

```diff
@@ -385,17 +389,102 @@ olink_one_non_parametric_posthoc <- function(df,
 
     if(test == "friedman"){
       message("Pairwise comparisons for Friedman test using paired Wilcoxon signed-rank test were performed")
-      p.hoc_val <- df %>%
+
+      #Not testing assays that have all NA:s in one level
+      #Every sample needs to have a unique level of the factor
+
+      nas_in_var <- character(0)
+
+      single_fixed_effects <- variable
+
+      for(effect in single_fixed_effects){
+
+        current_nas <- df %>%
+          dplyr::filter(!(OlinkID %in% npxCheck$all_nas)) %>%
+          dplyr::group_by(OlinkID, !!rlang::ensym(effect)) %>%
+          dplyr::summarise(n = dplyr::n(), n_na = sum(is.na(!!rlang::ensym(data_type))),.groups="drop") %>%
+          dplyr::filter(n == n_na) %>%
+          dplyr::distinct(OlinkID) %>%
+          dplyr::pull(OlinkID)
+
+        if(length(current_nas) > 0) {
+          nas_in_var <- c(nas_in_var, current_nas)
+          warning(paste0('The assay(s) ',
+                         current_nas,
+                         ' has only NA:s in atleast one level of ',
+                         effect,
+                         '. It will not be tested.'),
+                  call. = F)
+        }
+
+        number_of_samples_w_more_than_one_level <- df %>%
+          dplyr::group_by(SampleID) %>%
+          dplyr::summarise(n_levels = dplyr::n_distinct(!!rlang::ensym(effect), na.rm = T),.groups = "drop") %>%
+          dplyr::filter(n_levels > 1) %>%
+          nrow(.)
+
+        if (number_of_samples_w_more_than_one_level > 0) {
+          stop(paste0("There are ",
+                      number_of_samples_w_more_than_one_level,
+                      " samples that do not have a unique level for the effect ",
+                      effect,
+                      ". Only one level per sample is allowed."))
+        }
+      }
+
+      formula_string <- paste0(data_type, "~", paste(variable,collapse="*"))
+
+      #Get factors
+      fact.vars <- sapply(variable_testers, function(x) is.factor(df[[x]]))
+      fact.vars <- names(fact.vars)[fact.vars]
+
+      # add repeat measurement groups
+      df_nas_remove <- df %>%
+        dplyr::filter(!(OlinkID %in% npxCheck$all_nas)) %>%
+        dplyr::filter(!(OlinkID %in% nas_in_var))
+      # remove subject without complete data
+      ## number of the items in the subject
+
+      n_subject <- df_nas_remove %>%
+        dplyr::select(!!!rlang::syms(variable)) %>%
+        stats::na.omit() %>%
+        unique() %>%
+        pull() %>%
+        length()
+
+      subject_remove <- df_nas_remove %>%
+        dplyr::filter(!is.na(NPX)) %>%
+        dplyr::group_by(Assay, !!!rlang::syms(subject)) %>%
+        dplyr::summarize(n = n(),.groups = "drop") %>%
+        dplyr::filter(n == n_subject) %>%
+        dplyr::mutate(Friedman_remove = "no") %>%
+        dplyr::select(-n)
+
```

Comment:
subject_remove -> subject-Assay combos that are not missing levels of the variable